cluster_experiments/experiment_analysis.py

```diff
@@ -680,3 +682,226 @@ def analysis_point_estimate(self, df: pd.DataFrame, verbose: bool = False) -> fl
         """
         results_mlm = self.fit_mlm(df)
         return results_mlm.params[self.treatment_col]
+
+
+class SyntheticControlAnalysis(ExperimentAnalysis):
+    """
+    Class to run Synthetic control analysis. It expects only one treatment cluster.
+
+    Arguments:
+
+        target_col (str): The name of the column containing the variable to measure.
+        treatment_col (str): The name of the column containing the treatment variable.
+        treatment (str): The name of the treatment to use as the treated group.
+        cluster_cols (list): A list of columns to use as clusters.
+        hypothesis (str): One of "two-sided", "less", "greater" indicating the hypothesis.
+        time_col (str): The name of the column containing the time data.
+        intervention_date (str): The date when the intervention occurred.
+    Usage:
+
+    ```python
+    from cluster_experiments.experiment_analysis import SyntheticControlAnalysis
+    import pandas as pd
+    import numpy as np
+
+    dates = pd.date_range(start_date, end_date, freq="d")
+
+    users = [f"User {i}" for i in range(N)]
+
+    # Create a combination of each date with each user
+    combinations = list(product(users, dates))
+
+    target_values = np.random.normal(0, 1, size=len(combinations))
+
+    df = pd.DataFrame(combinations, columns=["user", "date"])
+    df["target"] = target_values
+
+
+    SyntheticControlAnalysis(
+        cluster_cols=["user"], time_col="date", transition_date="2022-01-15"
+    ).get_pvalue(df)
+
+    ```
+    """
+
+    def __init__(
+        self,
+        target_col: str = "target",
+        treatment_col: str = "treatment",
+        treatment: str = "B",
+        cluster_cols: Optional[List[str]] = None,
```

Comment:
is this really optional?

Reply:
if not, I suggest passing it as the first argument without a default

Reply:
Maybe we change it to:

cluster_cols: Optional[List[str]] = ['cluster']

so we keep the idea of always having a default

Reply:
wdyt about `cluster_cols: Union[List[str], Tuple[str]] = ('cluster',)`
and then doing `self.cluster_cols = list(cluster_cols)`?

Reply:
then I prefer to leave it empty haha